src/typing/typeloadCheck.ml

```diff
@@ -37,11 +37,12 @@ exception Build_canceled of build_state
 
 let is_generic_parameter ctx c =
 	(* first check field parameters, then class parameters *)
+	let name = snd c.cl_path in
 	try
-		ignore (List.assoc (snd c.cl_path) ctx.curfield.cf_params);
+		ignore(List.find (fun (n,_,_) -> n = name) ctx.curfield.cf_params);
```

Comment:
Aren't these the same as `lookup_param name ctx.type_params`?

Reply:
Uh, probably!